### PR TITLE
Adding official Facebook providers

### PR DESCRIPTION
**Pull request by TheCrow1213**

### CODE REVIEW

File: lib/oembed/providers.rb [hunk 1]
```diff
@@ -205,6 +205,24 @@ def add_official_provider(provider_class, sub_type=nil)
     Instagram << "https://www.instagram.com/p/*"
     add_official_provider(Instagram)
 
+    # Providers for Facebook Posts & Videos
+    # https://developers.facebook.com/docs/plugins/oembed-endpoints
+    FacebookPost = OEmbed::Provider.new('https://www.facebook.com/plugins/post/oembed.json/?url={format}')
```
Comment (metavida) [inline lib/oembed/providers.rb]: Based on [the Facebook oEmbed documentation](https://developers.facebook.com/docs/plugins/oembed-endpoints), it seems that they only support JSON responses. Furthermore, the `url=` portion of the URL is automatically added to the query string by [the `Provider#build` method](https://github.com/TheCrow1213/ruby-oembed/blob/v0.9.0/lib/oembed/provider.rb#L100-L103).

Please change both the `OEmbed::Provider.new` lines to use the following syntax:
```ruby
FacebookPost = OEmbed::Provider.new('https://www.facebook.com/plugins/post/oembed.json/', :json)
```

**Comment by metavida** [comment]
Thanks for the pull request!! There's one must-have change to both Facebook providers before this is ready to merge. Could you @ me once you've had a chance to make those tweaks?

**Comment by TheCrow1213** [comment]
Hey! Just saw your comments. Yes I will make the necessary tweaks and then I'll mention you :) Thanks a ton for reviewing!

**Comment by TheCrow1213** [comment]
@metavida Hey! Made the tweaks you pointed out. How does it look? Also added another url matcher for photos, found that I needed it in my app when I was testing stuff.

**Comment by metavida** [comment]
@TheCrow1213, thanks for adding these :sparkles: 

I did some local testing and the endpoints seem to be working great! I'm not 100% sure when I'll have time to release a new version of the gem (there's another pull request & and issue I'd like to include) but I'll do my best to make it sooner than later.

**Comment by TheCrow1213** [comment]
Ok, no rush :) whenever you have time. It was an easy addition to make!

**Comment by metavida** [comment]
@TheCrow1213 Alrighty, v0.10.0 is out & contains your new providers. Thanks again for the help!

**Pull request merged**